### DISCUSSION TITLE
Ability to sign request for AWS API

### DISCUSSION THREAD

**Discussion by camsellem** [grafana/grafana-infinity-datasource#473] [Ideas]
Hi

AWS API requires that all requests are signed using the following methodology - https://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html

This is something that is made transparently by AWS SDK and AWS CLI. Postman has also an option to sign requests for AWS API

It will be useful in order to query all AWS APis. For example, right now I'd like to build a dashboard with data from the Cost Explorer API.

**Comment by yesoreyeram** [comment]
Adding #501 and you can expect this in the next release

**Comment by camsellem** [comment]
This is awesome. The main use case for me is accessing AWS Cost Explorer
API ->
https://docs.aws.amazon.com/cost-management/latest/userguide/ce-api.html

**Comment by yesoreyeram** [comment]
AWS cost explorer example using infinity https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/505
